src/Microsoft.AspNet.Mvc.Xml/XmlDataContractSerializerOutputFormatter.cs

```diff
@@ -183,11 +185,31 @@ public override Task WriteResponseBodyAsync([NotNull] OutputFormatterContext con
                     obj = wrapperProvider.Wrap(obj);
                 }
 
-                var dataContractSerializer = CreateSerializer(wrappingType);
+                var dataContractSerializer = GetCachedSerializer(wrappingType);
                 dataContractSerializer.WriteObject(xmlWriter, obj);
             }
 
             return Task.FromResult(true);
         }
+
+        private DataContractSerializer GetCachedSerializer(Type type)
+        {
+            if (type == null)
```

Comment:
This is in the output formatter...so its CanWriteType?...can you return `false` if ResolveType returns null?